internal/assets/token_transfer.go

```diff
@@ -189,14 +190,14 @@ func (s *transferSender) resolveAndSend(ctx context.Context, method sendMethod)
 	return s.sendInternal(ctx, method)
 }
 
-func (s *transferSender) resolve(ctx context.Context) error {
+func (s *transferSender) resolve(ctx context.Context) (err error) {
 	// Resolve the attached message
 	if s.transfer.Message != nil {
-		sender, err := s.buildTransferMessage(ctx, s.namespace, s.transfer.Message)
+		s.msgSender, err = s.buildTransferMessage(ctx, s.namespace, s.transfer.Message)
```

Comment:
Seems like `msgSender` is only used within the context of this method and not reused if you call it twice. Should it just be a local instead of being on the struct?

Reply:
It is used to do the `Staged` write at the end